src/main/scala/org/bitcoins/core/util/Base58.scala

```diff
@@ -71,54 +71,24 @@ trait Base58 extends BitcoinSLogger {
   def encode(byte : Byte) : String = encode(Seq(byte))
 
   /**
-    * Encodes a pubkey hash to a base 58 address on the corresponding network
+    * If p2pkh, encodes the Sha256Hash160 of a public key to a base 58 address.
+    * If p2sh, encodes the Sha256Hash160 of a scriptPubKey to a base 58 address.
     * @param hash the result of Sha256(RipeMD160(pubkey))
-    * @param network the network on which this address is being generated for
+    * @param network the network on which this address is being generated for. (MainNet, TestNet3, RegTest)
     * @return
     */
-  def encodePubKeyHashToAddress(hash: Sha256Hash160Digest, network: NetworkParameters): Address = {
-    val versionByte: Byte = network.p2pkhNetworkByte
+  def encodeHashToAddress(hash: Sha256Hash160Digest, network: NetworkParameters, p2sh : Boolean): Address = {
```

Comment:
We already have this functionality https://github.com/bitcoin-s/bitcoin-s-core/blob/master/src/main/scala/org/bitcoins/core/protocol/Address.scala#L29